<launch>
	
	<!-- launch tabletop detection service -->
	<node name="table_object_detection_node" pkg="segbot_arm_perception" type="table_object_detection_node" output="screen">
	</node>
	
	<!-- launch tabletop detection service -->
	<node name="segbot_table_approach_as" pkg="segbot_arm_manipulation" type="segbot_table_approach_as" output="screen">
	</node>
	
	<!-- launch change detection service -->
	<node name="segbot_tabletop_grasp_as" pkg="segbot_arm_manipulation" type="segbot_tabletop_grasp_as" output="screen">
	</node>
	
	<!-- launch change detection service -->
	<node name="arm_lift_verify_as" pkg="segbot_arm_manipulation" type="arm_lift_verify_as" output="screen">
	</node>
	
	<!-- launch agile grasp -->
	
	<node name="find_grasps" pkg="agile_grasp" type="find_grasps" output="screen">
  	<param name="cloud_topic" value="/agile_grasp_demo/cloud" />
    <param name="cloud_frame" value="/xtion_camera_rgb_optical_frame" />
    <param name="cloud_type" value="0" />
    <param name="svm_file_name" value="$(find agile_grasp)/svm_032015_20_20_same" />
    <param name="num_samples" value="4000" />
    <param name="num_threads" value="4" />
    <param name="num_clouds" value="1" />
		<rosparam param="workspace"> [-1, 1, -1, 1, -1, 1.5] </rosparam>
    <rosparam param="camera_pose"> [0, 0.445417, 0.895323, 0.215, 
              1, 0, 0, -0.015, 
              0, 0.895323, -0.445417, 0.23, 
              0, 0, 0, 1] </rosparam>
    <param name="min_inliers" value="10" />
    <param name="plotting" value="2" />
    <param name="marker_lifetime" value="500" />
    <param name="hand_outer_diameter" value="0.10" />
    <param name="hand_depth" value="0.06" />
	</node>
	
	<!-- launch bwi_kr_execution -->
	
	<arg name="simulation" default="false" />
	<arg name="domain" default="$(find bwi_kr_execution)/domain/" />
	<arg name="executor" default="single_plan_executor_node" />

	<!-- available executors: single_plan_executor_node   any_plan_executor_node   learning_executor_node -->

	<param name="bwi_kr_execution/domain_directory" type="str" value="$(arg domain)" />

	<node name="bwi_kr" pkg="bwi_kr_execution" type="knowledge_reasoning_node">
	<param name="simulation" type="bool" value="$(arg simulation)" />
	</node>

	<node name="action_executor" pkg="bwi_kr_execution" type="$(arg executor)">
	<!-- launch-prefix="gdb -ex run -args" -->
	<param name="simulation" type="bool" value="$(arg simulation)" />
	</node>
	
</launch>
